### PR TITLE
fix: allow empty plugins for global rule

### LINKED ISSUE #1175: bug: if there is only one plugin in the plugin list, an error will be prompted when deleting it.
# Bug report

## Describe the bug

 if there is only one plugin in the plugin list, an error will be prompted when deleting it.

## How to Reproduce

1. Go to `plugin list` page
2. enable **one** plugin, like "prometheus"
3. deleting it by `Delete` button

image

```
Request URL: http://40.72.102.110:9000/apisix/admin/global_rules/1
Request Method: PUT
Request Body: {"id":"1","plugins":{}}
Response: {"code":10000,"message":"schema validate failed: (root): plugins is required","data":null,"request_id":"f4a2eaed-3698-44ca-b64c-223239336b8a"}
```

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
close #1175 

___
### Bugfix
- Description
please see #1175 for detail

- How to fix?
doesn't omit field `plugins` when it's an empty object.

### CODE REVIEW

File: api/test/e2e/global_rule_test.go [hunk 1]
```diff
@@ -234,7 +255,7 @@ func TestGlobalRule(t *testing.T) {
 			Object:        APISIXExpect(t),
 			Method:        http.MethodGet,
 			Path:          "/hello",
-			Query:         "name=;select%20from%20sys",
+			Query:         "file=root.exe",
```
Comment (juzhiyuan) [inline api/test/e2e/global_rule_test.go:258]: What's this? 😳

**Comment by starsz** [comment]
LGTM.

File: api/test/e2e/global_rule_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/global_rule_test.go:258]: it's query string for `uri-blocker` https://github.com/apache/apisix/blob/master/doc/plugins/uri-blocker.md

**Review by juzhiyuan — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]
after run this branch, I can delete the last one plugin in the global rule list.

**Pull request merged**